statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,410 @@
+    """
+
+    score_kwds = {} if score_kwds is None else score_kwds
+    hess_kwds = {} if hess_kwds is None else hess_kwds
+
+    if fit_kwds is None:
+        raise ValueError("_est_regularized_distributed currently " +
+                         "requires that fit_kwds not be None.")
+    else:
+        alpha = fit_kwds["alpha"]
+
+    if "L1_wt" in fit_kwds:
+        L1_wt = fit_kwds["L1_wt"]
+    else:
+        L1_wt = 1
+
+    n, p = mod.exog.shape
+    p_part = int(np.ceil((1. * p) / partitions))
+
+    beta_hat = mod.fit_regularized(**fit_kwds).params
+    grad = _gen_grad(mod, beta_hat, n, alpha, L1_wt, score_kwds)
+
+    X_beta = _gen_wdesign_mat(mod, beta_hat, hess_kwds)
+
+    gamma_hat_l = []
+    tau_hat_l = []
+    for pi in range(mnum * p_part, min((mnum + 1) * p_part, p)):
+        
+        gamma_hat = _gen_gamma_hat(X_beta, pi, p, n, alpha)
+        gamma_hat_l.append(gamma_hat)
+
+        tau_hat = _gen_tau_hat(X_beta, gamma_hat, pi, p, n, alpha)
+        tau_hat_l.append(tau_hat)
+
+    return beta_hat, grad, gamma_hat_l, tau_hat_l
+
+
+def _join_debiased(model_results_l, partitions, threshold=0):
+    """
+    joins the results from each run of _est_regularized_distributed
+    and returns the debiased estimate of the coefficients
+
+    Parameters
+    ----------
+    model_results_l : list
+        A list of tuples each one containing the beta_hat, grad,
+        gamma_hat and tau_hat values for each partition.
+    partitions : scalar
+        The number of partitions that the data will be split into.
+    threshold : scalar
+        The threshold at which the coefficients will be cut.
+    """
+
+    # TODO currently the way we extract p is roundabout, should be
+    # handled better but ideally not as an argument to _join_debiased
+
+    beta_hat_l = []
+    grad_l = []
+    gamma_hat_l = []
+    tau_hat_l = []
+
+    for r in model_results_l:
+        beta_hat_l.append(r[0])
+        grad_l.append(r[1])
+        gamma_hat_l.extend(r[2])
+        tau_hat_l.extend(r[3])
+
+    p = len(gamma_hat_l)
+
+    beta_mn = np.zeros(p)
+    for beta_hat in beta_hat_l:
+        beta_mn += beta_hat
+    beta_mn = beta_mn / partitions
+
+    grad_mn = np.zeros(p)
+    for grad in grad_l:
+        grad_mn += grad
+    grad_mn = -grad_mn / partitions
+
+    theta_hat = _gen_theta_hat(gamma_hat_l, tau_hat_l, p) 
+
+    beta_tilde = beta_mn + theta_hat.dot(grad_mn)
+
+    beta_tilde[np.abs(beta_tilde) < threshold] = 0
+
+    return beta_tilde
+#    return beta_tilde, beta_mn, theta_hat, grad_mn, gamma_hat_l, theta_hat.dot(grad_mn)
+
+
+def distributed_estimation(endog_generator, exog_generator, partitions,
+                           model_class=None, init_kwds=None, fit_kwds=None,
+                           estimation_method=None, estimation_kwds=None,
+                           join_method=None, join_kwds=None):
+    """
+    This functions handles a general approach to distributed estimation,
```

Comment:
First line of docstring should be one line.  Then blank line, then can have additional text that is longer than one line.